syntax = "proto3";


service MasterService {
    rpc RegisterMapper (RegisterMapperRequest) returns (RegisterMapperResponse) {}
    rpc RegisterReducer (RegisterReducerRequest) returns (RegisterReducerResponse) {}
  }

message RegisterMapperRequest {
    string ip = 1;
    int32 port = 2;
  }

message RegisterMapperResponse {
   enum Status {
    FAILED = 0;
    SUCCESS = 1;
  }
    string mapper_name = 1;
    Status status = 2;
}

message RegisterReducerRequest {
    string ip = 1;
    int32 port = 2;
  }

message RegisterReducerResponse {
   enum Status {
    FAILED = 0;
    SUCCESS = 1;
  }
    string reducer_name = 1;
    Status status = 2;
}